data.md

```diff
@@ -337,23 +347,84 @@ One needs to unname the `ValTypes` first before calling the `#take` or `#drop` f
+
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("09", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "t"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("0A", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "n"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("0D", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "r"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("22", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "\""
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("27", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "'"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("5C", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "\\"
+
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 8, SB +String Bytes2String(Int2Bytes(String2Base(substrString(S, IDX +Int 3, IDX +Int 7), 16), LE, Unsigned)))
+      requires IDX <Int lengthString(S) -Int 8
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "u"
+```
+
 Wasm memories can be initialized with a segment of data, sepcified as a string.
 The string considered to represent the sequence of UTF-8 bytes that encode it.
 The exception is for characters that are explicitly escaped which can represent bytes in hexadecimal form.
 To avoid dealing with these data strings in K, we use a list of integers as an initializer.
 
-**TODO:** Either convert from strings to integers directly in K or with a pre-processor.
+```k
+    syntax WasmString ::= ".WasmString"
+    syntax String     ::= #parseWasmString ( WasmString ) [function, functional, hook(STRING.token2string)]
+    syntax DataString ::= List{WasmString, ""}            [klabel(listWasmString)]
+```
+
+`DataString`, as is defined in the wasm semantics, is a list of `WasmString`s.
+`#concatDS` concatenates them together into a single string.
+The strings to connect are unescaped before concatenated, because unescape could remove the quote sign `"` before and after each substring.
```

Comment:
I don't understand what is meant by "could" here. Do you mean that the unescape always removes the quote sign?

Reply:
wrote a better description.